packages/node-cli/src/commands/actionresult/ActionResult.js

```diff
@@ -0,0 +1,100 @@
+/*
+** Copyright (c) 2019 Oracle and/or its affiliates.  All rights reserved.
+** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+*/
+'use strict';
+const assert = require('assert');
+
+const ERROR = "ERROR";
+const SUCCESS = "SUCCESS";
+
+class ActionResult {
+
+	constructor(build) {
+		this.validateBuild(build);
+
+		this._status = build.status;
+		this._data = build.data;
+		this._resultMessage = build.resultMessage;
+		this._errorMessages = build.errorMessages;
+	}
+
+	validateBuild(build) {
+		assert(build);
+		assert(build.status, "status is required when creating an ActionResult object.");
+		if (build.status === SUCCESS) {
+			assert(build.data, "data is required when ActionResult is a success.");
```

Comment:
That means withData should set successful status by default and there's no need for success() function